pkg/core/index.go

```diff
@@ -146,20 +106,22 @@ func (f *fileIndex) getFileIndex(pth string) (model.BundleEntries, error) {
 	return entries, nil
 }
 
-// downloadIndex downloads the i-th file index located by pather
+// downloadIndex asynchronously downloads the i-th file index located by pather
 func (f *fileIndex) downloadIndex(id string, pth string, index uint64, chans downloadBundleFileListChans) {
 	defer func() {
 		<-chans.concurrencyControl
 	}()
 
 	f.l.Info("downloading index file", zap.Uint64("current entry", index))
-	entries, err := f.getFileIndex(pth)
+	entries, err := f.getIndexFile(pth)
 	if err != nil {
 		chans.error <- err
 		return
 	}
 
-	// TODO(fred): to be checked out but I don't think the index value in this output is used beyond logging
+	// TODO(fred): nice - diamond does not use the idx field, but bundle_unpack does,
+	// we keep it for this method to be reusable by bundle_unpack at a later time.
+	// Ideally, we shouldn't need that index numbering.
```

Comment:
the `idx` struct field was added in case it's necessary to preserve order between requests and results – i.e. whether we need index numbering in the affirmative case is a consequence of preserving sort order.

so i suggest determining whether preserving sort order is desirable, making decision about circulating indices through the channels based on that determination.

Reply:
This investigation shall be conducted. From what I understand from the existing bundle functionality, this order preserving idea doesn't mean much, but I'll make sure later on.

Indeed, I recently drafted some experimental refactoring of this bundle download/upload stuff, which comes clearer now that I have to implement a more general functionality with 90% overlap. This draft is not published yet.